src/components/pages/Save/v2/reducer.ts

```diff
@@ -117,6 +117,14 @@ const reduce: Reducer<State, Action> = (state, action) => {
       };
     }
 
+    case Actions.SetModeType: {
+      const mode = action.payload;
```

Comment:
A bunch of stuff will have to happen here in the near future but it's ok for now